silkworm/silkrpc/json/types.cpp

```diff
@@ -70,24 +77,25 @@ std::size_t to_hex_no_leading_zeros(char* hex_bytes, silkworm::ByteView bytes) {
     }
     *dest = '\0';
 
-    return static_cast<size_t>(dest - hex_bytes);
+    return static_cast<size_t>(dest - hex_bytes.data());
 }
 
-std::size_t to_quantity(char* quantity_hex_bytes, silkworm::ByteView bytes) {
+std::size_t to_quantity(std::span<char> quantity_hex_bytes, silkworm::ByteView bytes) {
     return to_hex_no_leading_zeros(quantity_hex_bytes, bytes);
 }
 
-std::size_t to_quantity(char* quantity_hex_bytes, uint64_t number) {
+std::size_t to_quantity(std::span<char> quantity_hex_bytes, uint64_t number) {
     silkworm::Bytes number_bytes(8, '\0');
     silkworm::endian::store_big_u64(number_bytes.data(), number);
     return to_hex_no_leading_zeros(quantity_hex_bytes, number_bytes);
 }
 
-std::size_t to_quantity(char* quantity_hex_bytes, intx::uint256 number) {
+std::size_t to_quantity(std::span<char> quantity_hex_bytes, intx::uint256 number) {
     if (number == 0) {
         quantity_hex_bytes[0] = '0';
         quantity_hex_bytes[1] = 'x';
         quantity_hex_bytes[2] = '0';
+        quantity_hex_bytes[3] = '\0';
         return 3;
```

Comment:
This should be `4` to count also for the terminator character '\0'

Reply:
no the null is not counted

Reply:
TBH it all depends on the meaning of the return type. If we agree to make it `void` as returned value seems not to be used anywhere, we can simply forget.